examples/server/server.cpp

```diff
@@ -2192,6 +2197,38 @@ struct server_context {
             const std::string str_test = slot.generated_text.substr(pos);
             bool send_text = true;
 
+            if (slot.n_sent_text == 0 && slot.has_next_token && !slot.params.start_strings.empty()) {
+                size_t max_start_string_size = 0;
+                for (auto start_string: slot.params.start_strings) {
+                    max_start_string_size = std::max(max_start_string_size, start_string.size());
+                }
```

Comment:
We are re-running this loop every time a new token is generated, which waste computation power.

Instead, this should be pre-computed in `slot.params` and being reused